msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -704,4 +709,61 @@ public IAuthenticationResult acquireTokenSilent(@NonNull final AcquireTokenSilen
                 SINGLE_ACCOUNT_PCA_ACQUIRE_TOKEN_SILENT_WITH_PARAMETERS
         );
     }
+
+    @Override
+    protected DeviceCodeFlowCommandCallback getDeviceCodeFlowCommandCallback(@NonNull final DeviceCodeFlowCallback callback) {
+        return new DeviceCodeFlowCommandCallback<LocalAuthenticationResult, BaseException>() {
+
+            @Override
+            public void onUserCodeReceived(@NonNull final String vUri,
+                                           @NonNull final String userCode,
+                                           @NonNull final String message,
+                                           @NonNull final Date sessionExpirationDate) {
+                callback.onUserCodeReceived(vUri, userCode, message, sessionExpirationDate);
+            }
+
+            @Override
+            public void onTaskCompleted(LocalAuthenticationResult tokenResult) {
+                // Convert tokenResult to an AuthenticationResult object
+                final IAuthenticationResult convertedResult = AuthenticationResultAdapter.adapt(
+                        tokenResult);
+
+                // Type cast the interface object
+                final AuthenticationResult authResult = (AuthenticationResult) convertedResult;
+
+                // Persist the account in single account mode
+                persistCurrentAccount(tokenResult.getCacheRecordWithTenantProfileData());
+
+                callback.onTokenReceived(authResult);
+            }
+
+            @Override
+            public void onError(BaseException error) {
+                final MsalException msalException;
+
+                if (error instanceof ServiceException) {
+                    msalException = new MsalServiceException(
+                            error.getErrorCode(),
+                            error.getMessage(),
+                            ((ServiceException) error).getHttpStatusCode(),
+                            error
+                    );
+                } else {
+                    msalException = new MsalClientException(
+                            error.getErrorCode(),
+                            error.getMessage(),
+                            error
+                    );
+                }
+
+                callback.onError(msalException);
+            }
+
+            @Override
+            public void onCancel() {
+                // Do nothing
+                // No current plans for allowing cancellation of DCF
```

Comment:
Does this block other interactive requests?

Reply:
To my knowledge it doesn't, quickly tested this in MsalTestApp and AcquireToken can be called after the usercode is received for DCF and a token was received after sign in with AcquireToken